### PR TITLE
[EventFilter] Move private headers in plugins dir; merge plugins cc/h files

### PR DESCRIPTION
**Pull request by smuzaffar**
To avoid private header files exposed via package interface directory, this PR proposed to move the private headers in to plugins directory.
It also merges the `.cc and .h` files of plugins

This is part of cleanup discussed #34718

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: EventFilter/EcalRawToDigi/plugins/MatacqProducer.cc [hunk 1]
```diff
@@ -18,7 +18,7 @@
 #include "DataFormats/EcalDigi/interface/EcalMatacqDigi.h"
 #include "DataFormats/FEDRawData/interface/FEDNumbering.h"
 #include "DataFormats/FEDRawData/interface/FEDRawData.h"
-#include "EventFilter/EcalRawToDigi/interface/MatacqProducer.h"
+#include "EventFilter/EcalRawToDigi/src/MatacqProducer.h"
```
Comment (slava77) [inline EventFilter/EcalRawToDigi/plugins/MatacqProducer.cc:21]: this is a header file for `class MatacqProducer : public edm::EDProducer {`
We usually move them to /plugins and include without a path.
Yet a better option is to merge .h in .cc and drop it from SealModules.cc

Comment (smuzaffar) [reply EventFilter/EcalRawToDigi/plugins/MatacqProducer.cc]: done, I have merged the `.cc and .h` files for plugin and dropped `SealModules.cc`

**Comment by smuzaffar** [comment]
please test

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/34898	ddde684
- code changes are technical, in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show no relevant differences

**Comment by smuzaffar** [comment]
@rekovic, @cecilecaillol can you please review this?

**Comment by smuzaffar** [comment]
@rekovic, @cecilecaillol can you please review this?

**Comment by smuzaffar** [comment]
@cms-sw/l1-l2 , this is blocking the integration of build rules to enforce the private header usage check. Can you please review this?

**Comment by rekovic** [comment]
+1

**Review by perrotta — COMMENTED** [review]
A couple of duplicate includes. They could be better avoided if all includes are grouped together at the top of the file.
(Trivial fix that can also get postponed to a follow up PR)

**Comment by perrotta** [comment]
+1
- The duplicate includes were there even before merging the files...

**Comment by smuzaffar** [comment]
thanks @perrotta , I will open a PR to remove the duplicates

**Pull request merged**